std/digest/digest.d

```diff
@@ -992,3 +992,15 @@ unittest
     //necessary, result will have the correct length, but buf will still have it's original
     //length
 }
+
+@safe unittest
+{
+    // Test peek & length
+    import std.digest.crc;
+    auto hash = new WrapperDigest!CRC32();
+    assert(hash.length == 4);
+    hash.put(cast(const(ubyte[]))"The quick brown fox jumps over the lazy dog");
+    assert(hash.peek().toHexString() == "39A34F41");
+    ubyte[5] buf;
+    assert(hash.peek(buf).toHexString() == "39A34F41");
```

Comment:
I don't think we need to check whether it actually uses the buffer. The function needs to be horribly broken to fail such a test. Additionally the API does not guarantee that the input buffer is filled starting at 0 or that it even uses the input buffer. Using the returned slice is the only safe way to use the API.